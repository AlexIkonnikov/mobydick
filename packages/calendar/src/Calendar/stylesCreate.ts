import {Dimensions, StyleSheet} from 'react-native';
import {IThemeContext} from '@npm/mobydick-core/src/styles/theme/types';
import rem from '@npm/mobydick-core/src/styles/spaces/rem';

const {width: WIDTH} = Dimensions.get('window');

const stylesCreate = (theme: IThemeContext) => {
  const {spaces} = theme;

  return StyleSheet.create({
    calendar: {
      marginTop: spaces.Space12,
      width: WIDTH - spaces.Space8 * 2 - rem(16) * 2,
    },
    overlayStyle: {
      justifyContent: 'center',
    },
    contentCalendar: {
      width: WIDTH - spaces.Space8 * 2,
    },
  });
};

export default stylesCreate;
