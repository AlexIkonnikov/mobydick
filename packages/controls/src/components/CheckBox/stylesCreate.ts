import {IUseStylesTheme} from '@npm/mobydick-styles';
import {StyleSheet} from 'react-native';

const stylesCreate = (
  themeColors: IUseStylesTheme,
  disabled?: boolean,
  selected?: boolean,
) => {
  const {colors, spaces} = themeColors;

  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: colors.BgPrimary,
      opacity: disabled ? 0.4 : 1,
      maxWidth: '100%',
    },
    checkbox: {
      borderWidth: selected ? 0 : spaces.Space2,
      borderColor: colors.BorderNormal,
      width: spaces.Space20,
      height: spaces.Space20,
      borderRadius: spaces.Space4,
      margin: spaces.Space12,
    },
  });
};

export default stylesCreate;
