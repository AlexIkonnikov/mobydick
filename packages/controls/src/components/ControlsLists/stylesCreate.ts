import {StyleSheet} from 'react-native';
import {IThemeContext} from '@npm/mobydick-styles';

const stylesCreate = (_theme: IThemeContext, horizontal: boolean) =>
  StyleSheet.create({
    list: {
      flexDirection: horizontal ? 'row' : 'column',
    },
  });

export default stylesCreate;
