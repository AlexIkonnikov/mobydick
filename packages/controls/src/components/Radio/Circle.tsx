import React, {FC} from 'react';
import {Pressable, PressableProps, View} from '@npm/mobydick-core';

import {IRadioStyle} from './types';

const Circle: FC<IRadioStyle & PressableProps> = ({
  outerStyle,
  innerStyle,
  selected = false,
  ...rest
}) => {
  return (
    <Pressable style={outerStyle} {...rest}>
      {selected && <View style={innerStyle} />}
    </Pressable>
  );
};

export default Circle;
