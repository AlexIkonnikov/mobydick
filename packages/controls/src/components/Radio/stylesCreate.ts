import {StyleSheet} from 'react-native';
import {IThemeContext, rem} from '@npm/mobydick-styles';

const stylesCreate = (
  theme: IThemeContext,
  selected: boolean,
  disabled: boolean,
) => {
  const {colors, spaces} = theme;
  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: 'transparent',
      opacity: disabled ? 0.4 : 1,
    },
    circle: {
      width: rem(22),
      aspectRatio: 1,
      borderRadius: rem(11),
      backgroundColor: selected ? colors.ElementBase : 'transparent',
      borderColor: selected ? colors.ElementBase : colors.BorderNormal,
      borderWidth: spaces.Space2,
      justifyContent: 'center',
      alignItems: 'center',
      marginRight: spaces.Space12,
    },
    innerCircle: {
      width: rem(18),
      aspectRatio: 1,
      borderWidth: spaces.Space2,
      borderRadius: rem(9),
      borderColor: colors.BgPrimary,
    },
  });
};

export default stylesCreate;
