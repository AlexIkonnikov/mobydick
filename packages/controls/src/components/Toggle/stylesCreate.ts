import {IUseStylesTheme} from '@npm/mobydick-styles';
import {StyleSheet} from 'react-native';

const stylesCreate = (theme: IUseStylesTheme, disabled: boolean) => {
  const {colors, spaces} = theme;
  return StyleSheet.create({
    container: {
      width: 50,
      height: 30,
      borderRadius: spaces.Space20,
      padding: spaces.Space2,
      opacity: disabled ? 0.4 : 1,
    },
    switcher: {
      flex: 1,
      aspectRatio: 1,
      backgroundColor: colors.ElementWhite,
      borderRadius: 25,
    },
  });
};

export default stylesCreate;
