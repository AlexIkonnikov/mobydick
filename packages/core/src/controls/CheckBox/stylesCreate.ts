import {StyleSheet} from 'react-native';

import {IThemeContext} from '../../styles/types';
import rem from '../../styles/utils/rem';

const stylesCreate = (
  themeColors: IThemeContext,
  disabled = false,
  selected = false,
  width = rem(20),
  height = rem(20),
) => {
  const {colors, spaces} = themeColors;

  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: 'transparent',
      opacity: disabled ? 0.4 : 1,
      maxWidth: '100%',
    },
    checkbox: {
      borderWidth: selected ? 0 : spaces.Space2,
      borderColor: colors.BorderNormal,
      width: width,
      height: height,
      borderRadius: spaces.Space4,
    },
  });
};

export default stylesCreate;
