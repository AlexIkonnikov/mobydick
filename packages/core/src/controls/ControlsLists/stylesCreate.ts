import {StyleSheet} from 'react-native';

import {IThemeContext} from '../../styles/theme/types';

const stylesCreate = (_theme: IThemeContext, horizontal: boolean) =>
  StyleSheet.create({
    list: {
      flexDirection: horizontal ? 'row' : 'column',
    },
  });

export default stylesCreate;
