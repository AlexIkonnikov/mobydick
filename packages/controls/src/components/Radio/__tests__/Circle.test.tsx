import React from 'react';
import {render} from '@testing-library/react-native';
import {useStyles} from '@npm/mobydick-styles';
import {renderHook} from '@testing-library/react-hooks';

import Circle from '../Circle';
import stylesCreate from '../stylesCreate';

describe('@npm/mobydick-controls/Radio/Circle', () => {
  it('renders correctly', () => {
    const {result} = renderHook(() => useStyles(stylesCreate, false, false));
    const [styles] = result.current;
    const {toJSON} = render(
      <Circle innerStyle={styles.circle} outerStyle={styles.innerCircle} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly selected', () => {
    const {result} = renderHook(() => useStyles(stylesCreate, false, false));
    const [styles] = result.current;
    const {toJSON} = render(
      <Circle
        innerStyle={styles.circle}
        outerStyle={styles.innerCircle}
        selected
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
