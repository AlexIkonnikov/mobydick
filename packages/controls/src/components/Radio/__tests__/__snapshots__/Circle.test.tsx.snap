// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-controls/Radio/Circle renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "aspectRatio": 1,
      "borderColor": "#FFF",
      "borderRadius": 9,
      "borderWidth": 2,
      "width": 18,
    }
  }
/>
`;

exports[`@npm/mobydick-controls/Radio/Circle renders correctly selected 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "aspectRatio": 1,
      "borderColor": "#FFF",
      "borderRadius": 9,
      "borderWidth": 2,
      "width": 18,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "transparent",
        "borderColor": "rgba(32, 36, 45, 0.3)",
        "borderRadius": 11,
        "borderWidth": 3,
        "justifyContent": "center",
        "margin": 18.5,
        "width": 22,
      }
    }
  />
</View>
`;
